java/core/src/java/org/apache/orc/OrcFile.java

```diff
@@ -242,7 +242,7 @@ public static WriterVersion from(WriterImplementation writer, int val) {
             val + " for writer " + writer);
       }
       WriterVersion[] versions = values[writer.id];
-      if (val < 0 || versions.length < val) {
+      if (val < 0 || val >= versions.length) {
```

Comment:
If you don't mind, could you keep the previous style like the following?
```java
- if (val < 0 || val >= versions.length) {
- if (val < 0 || versions.length <= val) {
```

Reply:
I adjusted the order on purpose, comparing variables on the same side looks better to understand, that's how I feel.
Of course I don't mind changing it back, but I'd like to mention your opinion.

Reply:
6adff23, keep the previous style.  : ）